packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -200,15 +201,15 @@ export class Table extends Construct {
   private readonly secondaryIndexNames: string[] = [];
   private readonly nonKeyAttributes: string[] = [];
 
-  private tablePartitionKey?: Attribute;
+  private readonly tablePartitionKey: Attribute;
   private tableSortKey?: Attribute;
```

Comment:
I would take the opportunity to be consistent here & require you specify the schema at table creation... make `tableSortKey` `readonly`, too, and remove the `addSortKey` method...

Reply:
But sortKey is not required and I can see a use case for mutability post creation of the sort key

Reply:
I think

Reply:
What's the point? Usage of the table is going to imply the schema is known and determined (the sortKey is a required attribute, too). I just don't see any reason why primary and sort key should be handled distinctly. I call this uncalled-for future proofing.

Reply:
Also, it cannot be changed after the fact (update of `KeySchema` requires replacement), making it a dangerous mutation.

Reply:
okay

Reply:
okay, made sort key immutable